frame/session/src/lib.rs

```diff
@@ -328,11 +330,13 @@ impl<AId> SessionHandler<AId> for Tuple {
 		for_tuples!(
 			#(
 				let our_keys: Box<dyn Iterator<Item=_>> = Box::new(validators.iter()
-					.map(|k| (&k.0, k.1.get::<Tuple::Key>(<Tuple::Key as RuntimeAppPublic>::ID)
-						.unwrap_or_default())));
+					.filter_map(|k|
+						k.1.get::<Tuple::Key>(<Tuple::Key as RuntimeAppPublic>::ID).map(|k1| (&k.0, k1))
```

Comment:
We should probably forward here a `(ValidatorId, Option<Key>)` so that the called code is aware of all validators, but some are just not having a key set?

Reply:
Why? What would you expect calling code to do beyond reporting an error (which can be done here) and disregarding the validators? In any case this code is strictly more secure than before, so I don't think it should hold up the PR.

Reply:
What if the called code is only interested in the validator ids? I don't want to block this here, I just find it more secure to let the called code decide on what it wants to do. Currently it means that called code could have a different view on what is the actual set vs the session pallet (because we filtered out some validators).

Reply:
> What if the called code is only interested in the validator ids?

I don't think this has ever happened, but if it did then I would expect a different API should be provided.